File: popfragg.Tools.Codegen/Outputs/schemas/user-conflict-check-result.ts
import { z } from 'zod';

export const userConflictCheckResultSchema = z.object({
  steamIdConflict: z.string().optional(),
  nicknameConflict: z.string().optional(),
  emailConflict: z.string().optional(),
});

export type UserConflictCheckResult = z.infer<typeof userConflictCheckResultSchema>;

File: popfragg.Domain/DTOS/User/UserConflictCheckResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace popfragg.Domain.DTOS.User
{
    public class UserConflictCheckResult
    {
        public string? SteamIdConflict { get; set; }
        public string? NicknameConflict { get; set; }
        public string? EmailConflict { get; set; }

        public bool HasAnyConflict =>
            !string.IsNullOrEmpty(SteamIdConflict) ||
            !string.IsNullOrEmpty(NicknameConflict) ||
            !string.IsNullOrEmpty(EmailConflict);
    }
}
